### PR TITLE
Fix Tabbed/Stacked nested container front view

### PR DESCRIPTION
**Pull request by alexander-smoktal**
Fix Issue #407:
A Tabbed/Stacked nested in another Tabbed/Stacked container doesn't properly show the focused view

### CODE REVIEW

**Comment by alexander-smoktal** [comment]
Added fix. Also refactored a bit.

The fix is to add `handle.bring_to_front();` into `focus_on` function. As I understand if you focus on a view, you always want it to be in front of other windows?

**Comment by alexander-smoktal** [comment]
It's not that I don't like refactored code, but I rearranged code to be more readable for a side developer. If you don't like it, I'll revert.

File: src/layout/core/tree.rs [hunk 1]
```diff
@@ -216,37 +216,31 @@ impl LayoutTree {
...
-            ContainerType::View => try!(self.focus_on(id)),
+
+        match container_type {
+            ContainerType::View => self.focus_on(container_id)?,
             ContainerType::Container => {/* TODO implement this */},
             _ => return Err(
-                TreeError::UuidWrongType(id, vec!(ContainerType::View, ContainerType::Container)))
+                TreeError::UuidWrongType(container_id, vec!(ContainerType::View, ContainerType::Container)))
         }
-        self.tree.set_ancestor_paths_active(node_ix);
```
Comment (Timidger) [inline src/layout/core/tree.rs]: Why was this line removed? Did this interfere and cause an issue?

**Comment by Timidger** [comment]
Looks great, thanks for the refactoring. There's a fair amount of code that hasn't been brought up to date e.g using `?` instead of `try!`, and a lot of the code makes sense to me but not necessarily anyone else ;). 

Those small rewording/restructurings really help make it clear to anyone reading it what's going on without having to know the entire codebase and its conventions.

File: src/layout/core/tree.rs (hunk 1 above)
Comment (alexander-smoktal) [reply src/layout/core/tree.rs]: We have it inside `focus_on` call

Comment (Timidger) [reply src/layout/core/tree.rs]: Ah, good catch. Thanks for finding and removing the dup

**Comment by Timidger** [comment]
Thanks again for your contributions!
